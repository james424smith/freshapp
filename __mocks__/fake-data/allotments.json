[
  {
    "date": "27-06-2019",
    "beneficiary": "113282",
    "amount": "3500",
    "allotmentType": "Recurring",
    "currency": "USD"
  },
  {
    "date": "29-05-2019",
    "beneficiary": "113282",
    "amount": "3500",
    "allotmentType": "Recurring",
    "currency": "USD"
  }
]
