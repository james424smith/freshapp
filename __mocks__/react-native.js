import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
