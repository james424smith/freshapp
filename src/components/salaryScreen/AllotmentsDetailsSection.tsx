import React from "react";
import { View } from "react-native";
import AllotmentsDetailsComponent from "./AllotmentsDetailsComponent";
import { AllotmentsRecord } from "../../interfaces";
import valueOrDefault from "../../common/valueOrDefault";

type Props = {
  allotmentsRecordTable?: AllotmentsRecord[];
};

const splitAllotmentsRecordItems = (
  allotmentsItem: AllotmentsRecord,
  key: number
) => {
  return (
    <AllotmentsDetailsComponent
      key={key}
      dateText={allotmentsItem.date}
      beneficiaryText={allotmentsItem.beneficiary}
      amountText={allotmentsItem.amount}
      allotmentTypeText={allotmentsItem.allotmentType}
      currencyText={allotmentsItem.currency}
    />
  );
};

const AllotmentsDetailsSection = (props: Props) => {
  const { allotmentsRecordTable } = props;
  return (
    <View>
      {(valueOrDefault(
        allotmentsRecordTable,
        []
      ) as AllotmentsRecord[]).map(
        (allotmentItem: AllotmentsRecord, i: number) =>
          splitAllotmentsRecordItems(allotmentItem, i)
      )}
    </View>
  );
};

export default AllotmentsDetailsSection;
