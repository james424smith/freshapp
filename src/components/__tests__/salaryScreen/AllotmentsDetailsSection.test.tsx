import React from "react";
import AllotmentsDetailsSection from "../../salaryScreen/AllotmentsDetailsSection";
import { render, cleanup } from "@testing-library/react-native";
import allotments from "../../../../__mocks__/fake-data/allotments.json";

describe("AllotmentsDetailsSection snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui when allotmentTypeText", () => {
    const { toJSON } = render(
      <AllotmentsDetailsSection allotmentsRecordTable={allotments} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
