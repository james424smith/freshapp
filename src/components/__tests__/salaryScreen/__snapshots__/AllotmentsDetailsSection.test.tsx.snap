// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllotmentsDetailsSection snapshot test should render correct ui when allotmentTypeText 1`] = `
<View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 52.5,
          "marginTop": 37.5,
        }
      }
    >
      <View
        style={
          Object {
            "width": "40%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 28,
              },
            ]
          }
        >
          Beneficiary
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 29.5,
                "fontWeight": "bold",
              },
            ]
          }
        >
          113282
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "marginTop": 30,
          }
        }
      >
        <View
          style={
            Object {
              "width": "40%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 28,
                },
              ]
            }
          >
            Date
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            27-06-2019
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "35%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 28,
                },
              ]
            }
          >
            Amount
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            3500
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "35%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 28,
                },
              ]
            }
          >
            Type
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Recurring
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#78849E",
          "borderBottomWidth": 0.7,
          "marginLeft": 0,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 52.5,
          "marginTop": 37.5,
        }
      }
    >
      <View
        style={
          Object {
            "width": "40%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 28,
              },
            ]
          }
        >
          Beneficiary
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 29.5,
                "fontWeight": "bold",
              },
            ]
          }
        >
          113282
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "marginTop": 30,
          }
        }
      >
        <View
          style={
            Object {
              "width": "40%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 28,
                },
              ]
            }
          >
            Date
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            29-05-2019
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "35%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 28,
                },
              ]
            }
          >
            Amount
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            3500
          </Text>
        </View>
        <View
          style={
            Object {
              "width": "35%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 28,
                },
              ]
            }
          >
            Type
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#2A2E43",
                  "fontFamily": "Roboto-Regular",
                },
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 29.5,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Recurring
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#78849E",
          "borderBottomWidth": 0.7,
          "marginLeft": 0,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#2A2E43",
              "fontFamily": "Roboto-Regular",
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
